tests/core/manager/test_default_middlewares.py

```diff
@@ -0,0 +1,51 @@
+from web3.manager import (
+    RequestManager,
+)
+from web3.middleware import (
+    abi_middleware,
+    async_attrdict_middleware,
+    async_buffered_gas_estimate_middleware,
+    async_gas_price_strategy_middleware,
+    async_validation_middleware,
+    attrdict_middleware,
+    buffered_gas_estimate_middleware,
+    gas_price_strategy_middleware,
+    name_to_address_middleware,
+    pythonic_middleware,
+    request_parameter_normalizer,
+    validation_middleware,
+)
+
+
+def test_default_sync_middlwares(w3):
+    expected_middlewares = [
+        (request_parameter_normalizer, "request_param_normalizer"),
+        (gas_price_strategy_middleware, "gas_price_strategy"),
+        (name_to_address_middleware(w3), "name_to_address"),
+        (attrdict_middleware, "attrdict"),
+        (pythonic_middleware, "pythonic"),
+        (validation_middleware, "validation"),
+        (abi_middleware, "abi"),
+        (buffered_gas_estimate_middleware, "gas_estimate"),
+    ]
+
+    default_middlewares = RequestManager.default_middlewares(w3)
+
+    for x in range(len(default_middlewares)):
+        assert default_middlewares[x][0].__name__ == expected_middlewares[x][0].__name__
+        assert default_middlewares[x][1] == expected_middlewares[x][1]
+
+
+def test_default_async_middlwares():
```

Comment:
We have a test for this [here](https://github.com/ethereum/web3.py/blob/e0f17867b316aaf0f329c67c8e2d52fa25e6a6ee/tests/core/providers/test_async_http_provider.py#L68-L95). I only added the async version but perhaps we can make a similar one that is just as comprehensive, checking modules, etc as well?

I like the list approach, maybe a combination of these and that test? Thoughts?

Reply:
I like your testing as more comprehensive, but the location is less intuitive. If I'm making changes to `manager.py`, then I would look to update tests in the corresponding location in tests. Both locations seems overkill. I suppose if someone updates the default middleware without updating the provider test, it'll fail and they'll find it anyway.

Reply:
Happy to house these in a more appropriate place. I think I put it there because most of the test suite runs on `EthereumTesterProvider` and the middleware that is default for that is very different than the middleware that is default for a provider that connects to actual clients.

`test_base_provider.py` and `test_async_base_provider.py` might make more sense but the modules also only exist on the `Web3` instances so maybe these tests need to be separated as well.